pkg/util/runtimeconfig/manager.go

```diff
@@ -0,0 +1,172 @@
+package runtimeconfig
+
+import (
+	"flag"
+	"sync"
+	"time"
+
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var overridesReloadSuccess = promauto.NewGauge(prometheus.GaugeOpts{
+	Name: "cortex_overrides_last_reload_successful",
+	Help: "Whether the last config reload attempt was successful.",
+})
+
+// Loader loads the configuration from file.
+type Loader func(filename string) (interface{}, error)
+
+// ManagerConfig holds the config for an Manager instance.
+// It holds config related to loading per-tenant config.
+type ManagerConfig struct {
+	ReloadPeriod time.Duration `yaml:"period"`
+	LoadPath     string        `yaml:"file"`
+	Loader       Loader
+}
+
+// RegisterFlags registers flags.
+func (mc *ManagerConfig) RegisterFlags(f *flag.FlagSet) {
+	f.StringVar(&mc.LoadPath, "runtime-config.file", "", "File with the configuration that can be updated in runtime.")
+	f.DurationVar(&mc.ReloadPeriod, "runtime-config.reload-period", 10*time.Second, "How often to check runtime config file.")
+}
+
+// Manager periodically reloads the configuration from a file, and keeps this
+// configuration available for clients.
+type Manager struct {
+	cfg  ManagerConfig
+	quit chan struct{}
+
+	listenersMtx sync.Mutex
+	listeners    []chan interface{}
+
+	configMtx sync.RWMutex
+	config    interface{}
+}
+
+// NewRuntimeConfigManager creates an instance of Manager and starts reload config loop based on config
+func NewRuntimeConfigManager(cfg ManagerConfig) (*Manager, error) {
+	mgr := Manager{
+		cfg:  cfg,
+		quit: make(chan struct{}),
+	}
+
+	if cfg.LoadPath != "" {
+		if err := mgr.loadConfig(); err != nil {
+			// Log but don't stop on error - we don't want to halt all ingesters because of a typo
+			level.Error(util.Logger).Log("msg", "failed to load config", "err", err)
+		}
+		go mgr.loop()
+	} else {
+		level.Info(util.Logger).Log("msg", "config disabled")
```

Comment:
rutime config disabled.

Reply:
what about `"runtime config file not specified, reload disabled"` to explain "why" as well.

Reply:
or rather `runtime config disabled: file not specified`